File: JavaAngular/05_cliente_buscador/src/app/service/buscador.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Resultado } from '../model/Resultado';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class BuscadorService {

  constructor(private http:HttpClient) { }

  buscarResultados(tematica:string):Observable<Resultado[]>{
    let url:string = "http://localhost:7000/buscador/buscar";
    //llama asincrona
      return this.http.get<Resultado[]>(url+"?tematica="+tematica)
  }
}

File: Profesor/JavaEE-profe-servicio-rest/01_servicio_buscador/src/main/java/service/interfaces/BuscadorService.java
package service.interfaces;

import java.util.List;

import model.Resultado;

public interface BuscadorService {
	List<Resultado> buscar(String tematica);
	void agregar(Resultado resultado);
	List<Resultado> eliminarResultado(String url);
	Resultado actualizarDescripcion(String url, String nuevaDescripcion);
}
